### PR TITLE
Add a GitHub pull request template

### PR DESCRIPTION
**Pull request by andy-armstrong**
I've created a GitHub pull request template that will be automatically added to all PRs created going forward. For the content, I merged our recommend pull request template with the new standard one that Cale merged today:

https://github.com/edx/cookiecutter-django-ida/pull/21/files

FYI, you can read more about the GitHub functionality here:

https://github.com/blog/2111-issue-and-pull-request-templates

@AlasdairSwan @dsjen @dan-f Please review.

- [x] @bjacobel

### CODE REVIEW

**Comment by edx-webhook** [comment]
## PR Cover Letter
- [ ] A detailed description of the changes in the body of the PR
- [ ] If available, a link to the relevant JIRA ticket in the body of the PR
  - `[FEDX-XXXX](https://openedx.atlassian.net/browse/FEDX-XXXX)`

## Documentation Preview
- [ ] Build a preview site using ``gulp build-preview`` and add the generated link here

## Testing
- [ ] i18n
- [ ] Responsive design
- [ ] RTL

## Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] Code review: @Reviewer1
- [ ] Code review: @Reviewer2
- [ ] Documentation review: @...
- [ ] UX review: @...
- [ ] Accessibility review: @...
- [ ] Product review: @...

FYI: Tag anyone who might be interested in this PR here.

## Post-review
- [ ] Squash commits
- [ ] Publish packages/doc site

**Comment by dan-f** [comment]
The template contains a lot of information! The documentation is great, but I wonder how much of it belongs in the actual template itself. I'd rather not have to delete a bunch of text every time I create a new pull request.

I think that the `For the Submitter` and `Documentation considerations ` sections could be pushed out into a [`CONTRIBUTING.md`](https://github.com/blog/1184-contributing-guidelines). The remaining content would then constitute the actual template.

**Comment by dsjen** [comment]
Are specific users tagged for the non-code reviews too or are those just aspects that the reviewers should check off?

What about putting the explanatory text in a CONTRIB.md and reference that from the template?

**Comment by andy-armstrong** [comment]
Great points, both of you. I'll wait to see where things land with Cale's cookiecutter PR (https://github.com/edx/cookiecutter-django-ida/pull/22) and then I'll follow the same practices here.

**Comment by andy-armstrong** [comment]
@dan-f @dsjen @AlasdairSwan @bjacobel I've updated this PR based upon your feedback and using the UI Toolkit's template as a guide. Please review.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+TODO: In this section, tag specific reviewers you know will want to have a look at this PR. You can use the checklist below for organization:
+
+- [ ] @mention a reviewer here
+- [ ] @mention a reviewer here
+
+TODO: Consider not just code reviewers but also including reviewers from Docs, UX, Accessibility and Product, if applicable.
+
+If you've been tagged for review, please check your corresponding box once you've given the :+1:.
+
+We'll also automatically suggest reviewers for this PR based on the code it touches.
```
Comment (bjacobel) [inline .github/PULL_REQUEST_TEMPLATE.md]: We don't have this turned on for this repo yet, AFAIK (but we should).

**Comment by bjacobel** [comment]
LGTM, we we need to activate the mention bot though, do you have the right permissions to do that @andy-armstrong?

**Comment by andy-armstrong** [comment]
@bjacobel I forgot to mention that I turned on the mention-bot as part of this.

**Comment by bjacobel** [comment]
In that case :thumbsup:

**Comment by dsjen** [comment]
LGTM! 👍

**Comment by dan-f** [comment]
A few things:

1. If we're using the mention bot why do we need a reviewers section?
2. I personally find the "Manual Test Plan", "Testing Checklist", and "Non-testing Checklist" sections to be noisy. We should absolutely set testing expectations, but if our set of expectations don't change from PR to PR, then I'd rather not re-read these expectations in every single pull request. I think it's a shared responsibility between reviewer and author to make sure that the changes are sufficiently tested.

As for (2.) - one could argue the other direction that the constant reminders are useful and result in better tested code landing more consistently. So far I haven't noticed PR cover letters improving test coverage, but I won't rule it out.

**Comment by andy-armstrong** [comment]
@dan-f Good questions/points!

1. IMO the mention-bot isn't to specify a set of reviewers, but instead to give folks the heads up. Are you concerned for Open Source contributors that they won't know who to tag? Would it help to have a TODO describing how to handle this section?

2. I like the idea that the reviewer can see what kind of manual testing the author had done. It is too easy to have both sides just ignore this aspect IMO. I tend to delete the check list items that are not relevant from my PR before saving so it doesn't become noisy for the reviewer. Should we have a TODO describing this?

Regarding cover letters improving test coverage, I don't know of any repos other than the UI Toolkit that are using them yet. Do you mean the manual T&L ones didn't help? For me I found it made me think about it before I submitted the PR.

**Comment by dan-f** [comment]
Thanks for replying @andy-armstrong.

1. Yeah that is an important concern! I agree that it would be helpful to clarify in the template.
2. Based on experience I've always found that insufficient testing has been caught during code review regardless of whether there was a checklist on the cover letter. The major reason that I'm pushing back against checklists is that they seem to add a lot of noise to the cover letter. By increasing the vertical real estate of the cover letter I find it harder to locate key information.

It seems like the majority of folks I've talked to hold your opinion on the checklists, though. I don't want to block this from merging based on a personal preference, so 👍 once the tagging is cleared up.

**Comment by AlasdairSwan** [comment]
👍

**Pull request merged**